subsys/bluetooth/host/adv.c

```diff
@@ -269,6 +269,17 @@ void bt_le_ext_adv_foreach(void (*func)(struct bt_le_ext_adv *adv, void *data),
 #endif /* defined(CONFIG_BT_EXT_ADV) */
 }
 
+void bt_adv_reset_adv_flags(void)
+{
+#if defined(CONFIG_BT_EXT_ADV)
+	for (size_t i = 0; i < ARRAY_SIZE(adv_pool); i++) {
+		memset(adv_pool[i].flags, 0, BT_ADV_NUM_FLAGS);
```

Comment:
I'm wondering if we should just memset the whole `struct bt_le_ext_adv` element instead of just the `flags` member.
I'd prefer if we did this, if it doesn't break anything.
@alwa-nordic thoughts?

Reply:
I agree. The less the code has to "know", the better. Memseting the whole `adv_pool` variable is simplest. I have not verified that it's safe to do so, though.